libs/libarchfpga/src/read_xml_arch_file.cpp

```diff
@@ -3279,7 +3280,10 @@ static void ProcessEquivalentSiteDirects(pugi::xml_node Parent,
 
         int num_pins = from_pins.second - from_pins.first;
         for (int i = 0; i < num_pins; i++) {
-            directs_map[to_pins.first + i] = from_pins.first + i;
+            t_physical_pin phy_pin(from_pins.first + i);
+            t_logical_pin log_pin(to_pins.first + i);
+
+            directs_map.insert(log_pin, phy_pin);
```

Comment:
Should add an assertion on the return code of insert that the insert is successful (e.g. not duplicated).

Reply:
Done